packages/test-framework/src/fixture/pageObject/Checkout.ts

```diff
@@ -170,17 +170,22 @@ export class Checkout {
             addressAppendText,
         });
 
-        if (!isBillingAddressSame) await this.page.locator('label[for="sameAsBilling"]').click();
-        if (!shouldSaveAddress)
-            await this.page.locator('label[for="shippingAddress.shouldSaveAddress"]').click();
-
         await this.page.locator('#checkout-shipping-options').waitFor({ state: 'visible' });
 
         const firstShippingMethod = this.page
             .locator('#checkout-shipping-options .shippingOptions-container li')
             .nth(0);
 
         await firstShippingMethod.click();
+
+        if (!shouldSaveAddress) {
+            await this.page.getByText('Save this address in my address book.').click();
+        }
+
+        if (!isBillingAddressSame) {
+            await this.page.locator('label[for="sameAsBilling"]').click();
+        }
+
```

Comment:
This may be a side effect related to the changes.
The previous method of selecting checkboxes under the address form is no longer working.
However, simply moving the actions to the end resolves the problem.